### DISCUSSION TITLE
HydroDyn

### DISCUSSION THREAD

**Discussion by Southern-Gardenia8** [OpenFAST/openfast#1343] [Show and tell]
When I run FAST, the following error occurs in the system. Can you help me point out the problem?

OpenFAST input file heading:
    FAST Certification Test #25: NREL 5.0 MW Baseline Wind Turbine with OC4-DeepCwind semi
    configuration, for use in offshore analysis

Running ElastoDyn.
Nodal outputs section of ElastoDyn input file not found or improperly formatted.
Running InflowWind.
Running ServoDyn.
Running ServoDyn Interface for Bladed Controllers.
Using legacy Bladed DLL interface.
Running HydroDyn.
  WARNING:  The requested output channel is invalid: PRPSurge
  WARNING:  The requested output channel is invalid: PRPSway
  WARNING:  The requested output channel is invalid: PRPHeave
 Generating incident wave kinematics and current time history.
 Reading in WAMIT output with root name ".\../5MW_Baseline/HydroData/marin_semi".
 Computing radiation impulse response functions and wave diffraction forces.

FAST_InitializeAll:HydroDyn_Init:HydroDynInput_ProcessInitData: A requested output channel is
invalid
HydroDyn_Init: Error allocating space for the distributed dynamic pressure forces/moments array.

 FAST encountered an error during module initialization.
 Simulation error level: FATAL ERROR

 Aborting OpenFAST.

![问题1](https://user-images.githubusercontent.com/118819279/203323068-867740d4-aa62-4f36-b1cd-5806f7d94745.png)
![问题2](https://user-images.githubusercontent.com/118819279/203323076-73079e9a-cdef-4e1f-9d7f-fb114e87e7a3.png)

**Comment by jjonkman** [comment]
Dear @mhwzwy1314,

Regarding the "NaN state detected" error from MoorDyn (MD), I see that this error is preceded by a small rotation violation warning.  This suggests to me that your floating wind turbine model is unstable.  From your command prompt, it looks like you are simulating the NREL 5-MW baseline wind turbine atop the OC4-DeepCwind semisubmersible, which is a model that I would not expect to be unstable.  Did you change this model in some way?

Regarding the invalid "requested output channel" warnings, it looks like your HydroDyn write output names are not correct. Presumably you want to use HydroDyn output `B1Surge` etc. for the first potential-flow body. See the full list of available output channel names for the newest version of OpenFAST here: https://openfast.readthedocs.io/en/main/_downloads/3f19498a5dc774461e022b671ff01ec6/OutListParameters.xlsx.

Regarding the "allocating space" error, similar questions have been discussed before on our forum, e.g., see: http://forums.nrel.gov/t/problems-in-test-in-fast-8/1868.  If you are not already, I would recommend using the standard 64-bit, single-precision version of OpenFAST. Unless the problem is that you simply don’t have enough RAM available on your computer, in which case decreasing the discretization of the model (e.g., reducing the number of hydrodynamic analysis nodes) would probably help.

Best regards,

**Reply by Southern-Gardenia8** [reply]
Your help was very much appreciated.

**Comment by MujadidAldin** [comment]
Hi, My name is Aldin. I have a problem with the HydroDyn input file. I have converted MOSES output for frequency-dependent added mass, damping, and wave excitation like those in WAMIT output. Is there a solution for this?
![Screenshot (313)](https://user-images.githubusercontent.com/127412992/224044314-7a81f40a-3700-49d6-9f69-eb8552b707b2.png)

**Reply by jjonkman** [reply]
Dear @MujadidAldin,

Your issue sounds similar to one discussed recently here: https://github.com/OpenFAST/openfast/issues/1468.  As in that issue, my guess is there is something wrong with the data or format of your potential-flow input files (`PotFile`) from MOSES, which you have not shared.

Best regards,

**Reply by MujadidAldin** [reply]
I have checked my input data and it works now. Thank you
